web/scripts/database.ts

```diff
@@ -54,29 +54,51 @@ export function db_filter(indices: number[], rowcmpfn): number[] {
 }
 
 
-export function db_sort_numeric_minfirst(indices: number[], colidx): number[] {
+export function db_sort_numeric(indices: number[], colidx: number, minFirst: boolean): number[] {
     indices.sort(function (a, b) {
         let av = Number(opldb.data[a][colidx]);
         let bv = Number(opldb.data[b][colidx]);
+
+        let extreme = minFirst ? Number.MAX_VALUE : Number.MIN_VALUE;
         if (isNaN(av))
-            av = Number.MAX_VALUE;
+            av = extreme;
         if (isNaN(bv))
-            bv = Number.MAX_VALUE;
-        return av - bv;
+            bv = extreme;
+
+        let comparison = minFirst ? (av - bv) : (bv - av);
+
+        return comparison;
     });
     return indices;
 }
 
 
-export function db_sort_numeric_maxfirst(indices: number[], colidx): number[] {
+export function db_multicol_sort_numeric(indices: number[], col1idx: number, col2idx: number, minFirst: boolean): number[] {
     indices.sort(function (a, b) {
-        let av = Number(opldb.data[a][colidx]);
-        let bv = Number(opldb.data[b][colidx]);
-        if (isNaN(av))
-            av = Number.MIN_VALUE;
-        if (isNaN(bv))
-            bv = Number.MIN_VALUE;
-        return bv - av;
+        let av1 = Number(opldb.data[a][col1idx]);
+        let bv1 = Number(opldb.data[b][col1idx]);
+
+        let av2 = Number(opldb.data[a][col2idx]);
+        let bv2 = Number(opldb.data[b][col2idx]);
```

Comment:
In the common case where `av1 !== bv1`, performing loads for `av2` and `bv2` are wasted operations and quite expensive. It would be good to only execute these loads in the `if (av1 === bv1)` path below.